### PR TITLE
create `$$` as a lightweight, typed alternative to bonzo, query et al

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- adds `$$` as a convenience wrapper over DOM methods which also has types
- uses it
- bumps prettier

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No

## What is the value of this and can you measure success?

bonzo, qwery, bean etc are untyped. they also have complex APIs that are probably unnecessary now and would require a lot of time and effort to type. and types is what we want in TS land. in flow land, they were `any` for everything. which really is nothing.

there are some more up-to-date examples with types, e.g. [bliss](https://blissfuljs.com) and [cash](https://github.com/fabiospampinato/cash) but they offer more than we need and don't expect to work in a module bundler.

this is intended to be even lighter than they are, and we (@gtrufitt and me) anticipate that we (guardian developers) will borrow heavily from those two codebases as/if we need to add to more this lib. 

there doesn't seem a lot of point converting all instances of the older libs over, but this will lighten the load as and when code becomes TS/pulled out by @guardian/commercial-dev.

it includes a lint rule that will disallow using the older libs in TS too.

at some point, it may also move to https://github.com/guardian/libs if its useful outside frontend, or stay here if not.

### CODE REVIEW

**Comment by sndrs** [comment]
sweet, the tests pass locally but not on TC...

**Comment by sndrs** [comment]
then they passed! ¯\\_(ツ)_/¯

**Review by mxdvl — APPROVED** [review]
Looks great!

**Review by ioanna0 — APPROVED** [review]
👍

File: static/src/javascripts/projects/commercial/modules/ad-free-slot-remove.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import once from 'lodash/once';
+import { $$ } from '../../../lib/$$';
+import fastdom from '../../../lib/fastdom-promise';
```
Comment (ioanna0) [inline static/src/javascripts/projects/commercial/modules/ad-free-slot-remove.ts:3]: I think we can simplify the imports as: 

```suggestion
import { $$ } from 'lib/$$';
import fastdom from 'lib/fastdom-promise';
```

Comment (sndrs) [reply static/src/javascripts/projects/commercial/modules/ad-free-slot-remove.ts:3]: I removed all the magic paths from commercial stuff :) #23494 in prep for pulling it out, so commercial-core doesn't need to maintain them too

**Comment by mxdvl** [comment]
To make comparison easier, having a single commit handle the renaming from `*.js` to `*.ts` will help.

**Comment by gtrufitt** [comment]
Late review but looks great, nice one!

**Pull request merged**